scripts/clang-tidy-only-diff.sh

```diff
@@ -45,6 +45,12 @@ fi
 # -p Tells clang-tidy where to find the `compile_commands.json`.
 # `{}` specifies where `parallel` adds the command-line arguments.
 # `:::` separates the command `parallel` should execute from the arguments it should pass to the commands.
+{
+  while true; do
+    sleep 30
+    echo .
+  done
+} &
```

Comment:
i know this is aweful, but it seems like for some builds this step takes around 25min: https://circleci.com/gh/valhalla/valhalla/19414?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link. when that happens and no output is seen, CI just aborts the build which fails it.